packages/nextjs/components/scaffold-stark/Balance.tsx

```diff
@@ -16,21 +18,44 @@ type BalanceProps = {
  * Display (ETH & USD) balance of an ETH address.
  */
 export const Balance = ({ address, className = "", usdMode }: BalanceProps) => {
+  const price = useGlobalState((state) => state.nativeCurrencyPrice);
   const { targetNetwork } = useTargetNetwork();
+  const [balance, setBalance] = useState("0");
+  const [displayUsdMode, setDisplayUsdMode] = useState(
+    price > 0 ? Boolean(usdMode) : false,
+  );
 
-  const price = useGlobalState((state) => state.nativeCurrencyPrice);
-  const {
-    data: balance,
-    isError,
-    isLoading,
-  } = useBalance({
-    address,
+  const { data, isLoading, isError } = useContractRead({
+    address: ethContractAddress,
     watch: true,
+    abi: ethAbi,
+    functionName: "balanceOf",
+    args: [address as string],
+    blockIdentifier: "pending" as BlockNumber,
   });
 
-  const [displayUsdMode, setDisplayUsdMode] = useState(
-    price > 0 ? Boolean(usdMode) : false,
-  );
+  useEffect(() => {
+    // @ts-ignore
+    if (data && data.balance.low !== undefined) {
+      // @ts-ignore
+      const lowValueStr = data.balance.low.toString(); // Convert bigint to string
+      const len = lowValueStr.length;
+
+      let formattedValue;
+      if (len > 18) {
+        // More than 18 digits, so decimal point is not at the start
+        formattedValue =
+          lowValueStr.slice(0, len - 18) + "." + lowValueStr.slice(len - 18);
+      } else {
+        // Less than or equal to 18 digits, need to pad with zeros
+        formattedValue = "0." + "0".repeat(18 - len) + lowValueStr;
+      }
+
+      // Round the string number to 4 decimal places after moving the decimal point
+      const roundedValue = parseFloat(formattedValue).toFixed(4);
+      setBalance(roundedValue); // Update state with formatted value
+    }
+  }, [data]);
```

Comment:
please use something like [here](https://github.com/Quantum3-Labs/scaffold-stark-2/blob/1adbff390d70de734e01116fd8a86d9807cf80e3/packages/nextjs/utils/scaffold-stark/contract.ts#L321C1-L322C1) for parsing the `uint256` struct back to BigNumber